src/Nox.Abstractions/Domain/IRepository.cs

```diff
@@ -0,0 +1,20 @@
+﻿
+namespace Nox.Domain;
+
+public interface IRepository
+{
+    /// <summary>
+    /// Deletes Entity
+    /// </summary>
+    void Delete<T>(T entity) where T : IEntity;
+
+    /// <summary>
+    /// Deletes Owned Entity
+    /// </summary>
+    void DeleteOwned<T>(T entity) where T : IOwnedEntity;
+
+    /// <summary>
+    /// Deletes a range of Owned Entities
+    /// </summary>
+    void DeleteOwnedRange<T>(IEnumerable<T> entities) where T : IOwnedEntity;
```

Comment:
The name and the parameters, overload function, are sufficient 
```suggestion
    void DeleteOwned<T>(IEnumerable<T> entities) where T : IOwnedEntity;
```

Reply:
Updated in https://github.com/NoxOrg/Nox.Generator/pull/1206